road_safety_scanner.py

```diff
@@ -181,24 +190,49 @@ def getFilters(self):
         return userQuery, authorName, publishDate, keyWordsList, setting
 
     def selectAI(self):
-        """Set the API key for the GPT model."""
+        """Set the API key for the GPT model and update the current AI label."""
 
         clicked_button = self.sender()
-        print(f"AI set to: {clicked_button.text()}")  # Debug output to confirm the selection
+        selected_ai = clicked_button.text()  # Get the text of the clicked button (AI name)
+        print(f"AI set to: {selected_ai}")  # Debug output to confirm the selection
```

Comment:
Can also get rid of this Debug statement, but other than this I think this should be ok